vignettes/target_types.Rmd

```diff
@@ -0,0 +1,49 @@
+---
+title: "Target type guidelines"
+author: "David Watkins"
+date: "3/25/2019"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Target types}
+  %\VignetteEngine{knitr::rmarkdown}
+  \usepackage[utf8]{inputenc}
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE)
+```
+
+`scipiper` targets can be indicator files with a [shared cache](shared_cache.html), local data files with or without local indicator files, or simply R objects.  This vignette provides some guidelines for when to use these different patterns.
+
+## Objectives
+There are two main goals to keep in mind when deciding what style of target to use:
+ 
+ - **Minimize time spend unnecessarily rebuilding targets** --- i.e., those that have already been built locally or by collaborators
+ - **Keeping the pipeline simple** --- caching can save lots of build time, but also adds to the code that has to be maintained, can clutter the repo with build and indicator files, etc.
+ 
+ These two can sometimes be in tension, but usually it will be clear which is more important for a given case.
+ 
+## Key factors to consider
```

Comment:
These factors apply specifically to the question of whether something should be in the shared cache or not. There's also the question of whether something local-only should be an object or a file. Ideally this vignette would discuss both questions because they so often come up together.